lib/tnOption.js

```diff
@@ -0,0 +1,81 @@
+var Client = require("./client");
+var TN_OPTION_PATH = "tnoptions";
+
+function TnOption(){
+}
+
+TnOption.get = function(client, orderId, callback){
+  if(arguments.length === 2){
+    callback = orderId;
+    orderId = client;
+    client = new Client();
+  }
+  client.makeRequest("get", client.concatAccountPath(TN_OPTION_PATH) + "/" + orderId, function(err, order){
+    if(err){
+      return callback(err);
+    }
+    if (order.tnOptionGroups && order.tnOptionGroups.tnOptionGroup) {
+      order.tnOptionGroups = order.tnOptionGroups.tnOptionGroup;
+      if (!Array.isArray(order.tnOptionGroups)) {
+        order.tnOptionGroups = [order.tnOptionGroups];
+      }
+    }
+    order.tnOptionGroups.forEach((item) => {
+      if (item.telephoneNumbers && item.telephoneNumbers.telephoneNumber) {
+        item.telephoneNumbers = item.telephoneNumbers.telephoneNumber
+        if (!Array.isArray(item.telephoneNumbers)) {
+          item.telephoneNumbers = [item.telephoneNumbers];
+        }
+      }
+    });
+    callback(null, order);
+  });
+};
+
+TnOption.list = function(client, query, callback){
+  if(arguments.length === 2){
+    callback = query;
+    query = client;
+    client = new Client();
+  }
+  client.makeRequest("get", client.concatAccountPath(TN_OPTION_PATH), query, function(err,res){
+    if(err){
+      return callback(err);
+    }
+    var items = res.tnOptionOrderSummary || res.tnOptionOrder || [];
+    if(!Array.isArray(items)){
+      items = [items];
+    }
+    callback(null, items);
+  });
+}
+
+TnOption.create = function(client, order, callback){
+  if(arguments.length === 2){
+    callback = order;
+    order = client;
+    client = new Client();
+  }
+  order.tnOptionGroups = {tnOptionGroup: order.tnOptionGroups}
+  order.tnOptionGroups.tnOptionGroup.forEach((group) => {
+    group.telephoneNumbers = {telephoneNumber: group.telephoneNumbers}
+  });
+
+  var request = client.createPostRequest(client.concatAccountPath(TN_OPTION_PATH), {tnOptionOrder: order});
+  request.buffer().then(res =>{
+    if(res.ok && res.headers.location){
+      Client.getIdFromLocationHeader(res.headers.location, function(err, id){
+        if(err){
+          return callback(err);
+        }
+        TnOption.get(client, id, callback);
```

Comment:
I don't think this call is necessary. Why can't you just return the response from the post request?

Reply:
That's weird. I saw the other parts of the SDK doing it this way, so I just figured there was a reason for it. Fixed.

Reply:
ah that's strange, yeah I don't think that's necessary